r/tests/testthat/test-dplyr-funcs-string.R

```diff
@@ -433,7 +433,24 @@ test_that("str_replace and str_replace_all", {
       collect(),
     df
   )
-
+  expect_error(
+    arrow_table(df) %>%
+      transmute(x = call_binding("str_replace_all", x, c("F" = "_", "b" = ""))) %>%
+      collect(),
+    regexp = "`pattern` must be a length 1 character vector",
+  )
+  expect_error(
+    arrow_table(df) %>%
+      transmute(x = call_binding("str_replace_all", x, c("F", "b"), c("_", ""))) %>%
+      collect(),
+    regexp = "`pattern` must be a length 1 character vector",
+  )
+  expect_error(
+    arrow_table(df) %>%
+      transmute(x = call_binding("str_replace_all", x, c("F"), c("_", ""))) %>%
+      collect(),
+    regexp = "`replacement` must be a length 1 character vector",
+  )
```

Comment:
One tiny last change to suggest, and then this will be ready to merge!  You don't need the `arrow_table()` etc stuff here as `call_binding()` can be called on its own.  What is needed instead is to create a field reference so `call_binding()` has something to refer to in the expression it creates, and then `expect_error()` can just wrap `call_binding()`.

For example, in that final test, you'll need something a bit shorter, like this:

```
  x <- Expression$field_ref("x")
  expect_error(
    call_binding("str_replace_all", x, c("F"), c("_", ""))),
    regexp = "`replacement` must be a length 1 character vector"
  )
```

Reply:
Cool.  I'll be honest.  I really am lost in this whole `call_bindings`/`Expression$field_ref("x")` stuff.  I guess thats what I get for being a field biologist dabbling in computer science.  Thanks for walking me through this stuff!

Reply:
You've dived into a really tricky bit of the codebase, but you've done a great job!  A lot of my own early PRs to arrow involved drawing analogies between different bits of the codebase, but not understanding exactly what was going on (and this is still the case for any of my PRs which involve any C++).

Congratulations on your first PR to Arrow! Once the CI passes I'll merge it.  Welcome to the project :)